apps/web/src/pages/templates/components/email-editor/TextRowContent.tsx

```diff
@@ -63,13 +172,61 @@ export function TextRowContent({ blockIndex }: { blockIndex: number }) {
                 const html = e.target.innerHTML;
                 field.onChange(html);
                 checkPlaceholderVisibility(html);
+                if (showAutoSuggestions) {
+                  const variableTyped = diffStrings(contentSnapshot, html);
+                  setVariableQuery(variableTyped);
+                }
```

Comment:
As soon as the user types {{ and enables the autosuggestions dropdown functionality
we take a snapshot of the text content
and when the user starts typing the variable name
we compare the new text content with the old snapshot to get the variable typed text
this is then further used to filter suggestions